src/main/java/de/hpi/bpt/argos/storage/hierarchy/EntityHierarchyNodeImpl.java

```diff
@@ -0,0 +1,67 @@
+package de.hpi.bpt.argos.storage.hierarchy;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * {@inheritDoc}
+ * This is the implementation.
+ */
+public class EntityHierarchyNodeImpl implements EntityHierarchyNode {
+
+	private long id;
+	private List<EntityHierarchyNode> children;
+
+	/**
+	 * This constructor initializes the id member with the given parameter.
+	 * @param id - the id to be set
+	 */
+	public EntityHierarchyNodeImpl(long id) {
+		this.id = id;
+		children = new ArrayList<>();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public long getId() {
+		return id;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void addChild(EntityHierarchyNode child) {
+		children.add(child);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public boolean hasChildren() {
+		return !children.isEmpty();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public EntityHierarchyNode findEntity(long id) {
```

Comment:
This only looks into the children of this node, so maybe rename that method to findChildEntity() or similar? Otherwise I'd only expect the method to return null if the id I'm looking for doesn't exist at all.